pkg/operatorsource/handler.go

```diff
@@ -2,50 +2,88 @@ package operatorsource
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/kube"
+	"github.com/operator-framework/operator-marketplace/pkg/operatorsource/phase"
 	"github.com/operator-framework/operator-sdk/pkg/sdk"
 	"github.com/sirupsen/logrus"
 )
 
+// NewHandlerWithParams returns a new Handler.
+func NewHandlerWithParams(factory PhaseReconcilerFactory, kubeclient kube.Client, transitioner phase.Transitioner) Handler {
+	return &handler{
+		factory:      factory,
+		kubeclient:   kubeclient,
+		transitioner: transitioner,
+	}
+}
+
 // Handler is the interface that wraps the Handle method
 //
-// Handle handles a new event associated with OperatorSource type
+// Handle handles a new event associated with OperatorSource type.
+//
+// ctx represents the parent context.
+// event encapsulates the event fired by operator sdk.
 type Handler interface {
 	Handle(ctx context.Context, event sdk.Event) error
 }
 
+// handler implements the Handler interface
 type handler struct {
-	reconciler Reconciler
+	factory      PhaseReconcilerFactory
+	kubeclient   kube.Client
+	transitioner phase.Transitioner
 }
 
 func (h *handler) Handle(ctx context.Context, event sdk.Event) error {
-	opsrc := event.Object.(*v1alpha1.OperatorSource)
-
-	if event.Deleted {
-		logrus.Infof("No action taken, object has been deleted [type=%s object=%s/%s]",
-			opsrc.TypeMeta.Kind, opsrc.ObjectMeta.Namespace, opsrc.ObjectMeta.Name)
-
-		return nil
+	in, ok := event.Object.(*v1alpha1.OperatorSource)
+	if !ok {
+		return fmt.Errorf("casting failed, wrong type provided")
 	}
 
-	reconciled, err := h.reconciler.IsAlreadyReconciled(opsrc)
+	logger := getLoggerWithOperatorSourceFields(in)
+	ctxWithLogger := context.WithValue(ctx, "logger", logger)
```

Comment:
logger is no longer being propagated via `context`. 
I answered your question in haste, that we can use `context.Values` to propagate logger. `context.Value` is ideal to propagate request scoped data like `correlation-id` and such so that it can be passed to other layers or even across API boundaries. 
But a contextual logger, although it is tied to a reconcile event, should not be passed via context. 
I have added a member variable `logger` for `reconciler` struct types.